### PR TITLE
Move optimize methods to overload operators and optimize Strassen

### PR DESCRIPTION
**Pull request by toor1245**

### CODE REVIEW

File: src/MatrixDotNet/Extensions/Performance/MultiplyMatrixUnsafe.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            a.SplitMatrix(out var a11, out var a12, out var a21, out var a22);
+            b.SplitMatrix(out var b11, out var b12, out var b21, out var b22);
+
+            var p1 = Task.Run(() => MultiplyStrassen(a11 + a22, b11 + b22));
+            var p2 = Task.Run(() => MultiplyStrassen(a21 + a22, b11));
+            var p3 = Task.Run(() => MultiplyStrassen(a11, b12 - b22));
+            var p4 = Task.Run(() => MultiplyStrassen(a22, b21 - b11));
+            var p5 = Task.Run(() => MultiplyStrassen(a11 + a12, b22));
+            var p6 = Task.Run(() => MultiplyStrassen(a21 - a11, b11 + b12));
+            var p7 = Task.Run(() => MultiplyStrassen(a12 - a22, b21 + b22));
```
Comment (Dilorfin) [inline src/MatrixDotNet/Extensions/Performance/MultiplyMatrixUnsafe.cs:37]: What do you think about arrays?

File: tests/MatrixDotNet.PerformanceTesting/MathExtension/AddTest.cs [hunk 2]
```diff
@@ -18,9 +18,9 @@ public void Setup()
         }
 
         [Benchmark]
-        public Matrix<double> AddUnsafe()
+        public Matrix<double> AddOptimizationUnsafe()
         {
-            return Matrix<double>.Plus(_matrix1,_matrix2);
+            return _matrix1 + _matrix2;
```
Comment (Dilorfin) [inline tests/MatrixDotNet.PerformanceTesting/MathExtension/AddTest.cs:23]: We have this code on lines 27-30 in this file

Comment (toor1245) [reply tests/MatrixDotNet.PerformanceTesting/MathExtension/AddTest.cs:23]: ok I remove it

File: src/MatrixDotNet/Extensions/Performance/MultiplyMatrixUnsafe.cs (hunk 1 above)
Comment (toor1245) [reply src/MatrixDotNet/Extensions/Performance/MultiplyMatrixUnsafe.cs:37]: I think it is will be less readable and there will be no effect in performance

Comment (toor1245) [reply src/MatrixDotNet/Extensions/Performance/MultiplyMatrixUnsafe.cs:37]: but we used params in `Task.WaitAll` which creates implicit array

**Review by Dilorfin — APPROVED** [review]

**Review by Dilorfin — APPROVED** [review]

**Pull request merged**